wrapper/src/main/java/software/amazon/jdbc/dialect/PgDialect.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.dialect;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Set;
+import software.amazon.jdbc.exceptions.ExceptionHandler;
+import software.amazon.jdbc.exceptions.PgExceptionHandler;
+
+public class PgDialect implements Dialect {
+
+  private static PgExceptionHandler pgExceptionHandler;
+
+  @Override
+  public int getDefaultPort() {
+    return 5432;
+  }
+
+  @Override
+  public ExceptionHandler getExceptionHandler() {
+    if (pgExceptionHandler == null) {
+      pgExceptionHandler = new PgExceptionHandler();
+    }
+    return pgExceptionHandler;
+  }
+
+  @Override
+  public String getHostAliasQuery() {
+    return "SELECT CONCAT(inet_server_addr(), ':', inet_server_port())";
+  }
+
+  @Override
+  public String getServerVersionQuery() {
+    return "SELECT VERSION()";
+  }
+
+  @Override
+  public boolean isConnectionDialect(Connection connection) {
```

Comment:
This method name isConnectionDialect is a bit confusing to me

Reply:
The idea is to let dialect to identify a connected database by itself. In case when the driver can't identify a dialect (unknown protocol is used?) or when there're doubts which database flavour it is (vanila Mysql or Auror Mysql), it can be a solution. It's not in use for now. I'd keep it for some time and see where it goes.